### ISSUE TITLE
Extending the model builder with a webcam support

### ISSUE THREAD

**Issue by acostin1** [tensorflow/tfjs-core#605]
Hi,
I am playing with deeplearn.js, impressive work so far and it's really easy to explain some of the concepts to my son - but I am not an expert in neural networks not in HTML anymore  (and polymer and typescript are raising the bar :).

I am trying to enhance the model builder demo, have it include a webcam and then capture the webcam content and infer it.

So far, I have a const image = Array3D.fromPixels(element, 3) captured from a video element I have added to the demo (inspired by the imagenet demo). However, in the Imagenet/Squeezenet demo, there is a predictWithActivation() method that doesn't exist for the dynamic model from the model-builder folder.

I just don't know how to pass it to the GraphRunner and have it evaluated, also not sure how to extract the labels.

Any chance for some code pointers there?

Thanks

**Comment by acostin1** [comment]
Made some progress.

const element = this.webcamVideoElement;
const image = Array3D.fromPixels(element, 3) as NDArray;
this.cameraContent = ENV.math.cast(image, "float32");

Then I use it in the getTestData method.

const [images, labels] = this.dataSet.getData() as [NDArray[], NDArray[]];
for (let i = 0; i < 15; i++)
  images[images.length - 15 + i] = this.cameraContent;
return [images.slice(images.length - 15), labels.slice(labels.length - 15)];

However seems like I need to normalize the inputs as they are RGB - and I have to select the dropdown to be "None" for nomalization for my images to appear. Should I normalize them to -1,1? What's the formula?

**Comment by nsthorat** [comment]
Hi @acostin1 - apologies for the delays :)

We're going to make a much more powerful model-builder in the coming months (which can take advantage of the webcam).
